File: androidApp/features/diary/src/main/kotlin/com/velkonost/getbetter/android/features/diary/notes/components/createnewnote/tags/TagItem.kt
package com.velkonost.getbetter.android.features.diary.notes.components.createnewnote.tags

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.RowScope
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.ColorFilter
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.velkonost.getbetter.shared.features.diary.model.TagUI
import com.velkonost.getbetter.shared.resources.SharedR
import dev.icerock.moko.resources.compose.colorResource
import dev.icerock.moko.resources.compose.painterResource

@Composable
fun RowScope.TagItem(
    modifier: Modifier = Modifier,
    tag: TagUI,
    onTagDelete: (String) -> Unit
) {
    val interactionSource = remember { MutableInteractionSource() }

    Row(
        modifier = modifier
            .height(24.dp)
            .align(Alignment.CenterVertically)
            .background(
                color = colorResource(resource = SharedR.colors.button_gradient_start),
                shape = MaterialTheme.shapes.small
            )
            .padding(
                start = 6.dp,
                end = 6.dp,
                top = 3.dp,
                bottom = 3.dp
            ),
    ) {
        Text(
            modifier = modifier
                .height(24.dp)
                .align(Alignment.CenterVertically),
            text = tag.text,
            textAlign = TextAlign.Center,
            color = colorResource(resource = SharedR.colors.text_secondary),
            style = MaterialTheme.typography.bodyMedium
        )

        Image(
            modifier = modifier
                .padding(start = 4.dp)
                .align(Alignment.CenterVertically)
                .size(14.dp)
                .clickable(
                    interactionSource = interactionSource,
                    indication = null
                ) {
                    onTagDelete.invoke(tag.text)
                },
            painter = painterResource(imageResource = SharedR.images.ic_close),
            contentDescription = null,
            colorFilter = ColorFilter.tint(
                color = colorResource(resource = SharedR.colors.text_secondary)
            )
        )
    }
}

File: iosApp/iosApp/Features/Diary/notes/components/createnewnote/tags/TagItem.swift
//
//  TagItem.swift
//  iosApp
//
//  Created by velkonost on 26.10.2023.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import SharedSDK

struct TagItem : View {
    
    let tag: TagUI
    let onTagDelete: (String) -> Void
    
    init(tag: TagUI, onTagDelete: @escaping (String) -> Void) {
        self.tag = tag
        self.onTagDelete = onTagDelete
    }
    
    var body: some View {
        HStack(spacing: 0) {
            Text(tag.text)
                .style(.bodyMedium)
                .foregroundColor(.textSecondaryTitle)
                .frame(height: 30, alignment: .center)
            
            Image(uiImage: SharedR.images().ic_close.toUIImage()!)
                .resizable()
                .renderingMode(.template)
                .foregroundColor(.textSecondaryTitle)
                .scaledToFit()
                .frame(width: 18, height: 18, alignment: .center)
                .padding(.leading, 4)
                .onTapGesture {
                    onTagDelete(tag.text)
                }
        }
        .padding(.init(top: 3, leading: 6, bottom: 3, trailing: 6))
        .frame(height: 30)
        .background(
            RoundedRectangle(cornerRadius: 8)
                .fill(Color.buttonGradientStart)
        )
    }
}
